src/sentry/integrations/vsts/issues.py

```diff
@@ -20,31 +20,55 @@ class VstsIssueSync(IssueSyncMixin):
     issue_fields = frozenset(['id', 'title', 'url'])
     done_categories = frozenset(['Resolved', 'Completed'])
 
-    def get_create_issue_config(self, group, **kwargs):
-        fields = super(VstsIssueSync, self).get_create_issue_config(group, **kwargs)
+    def get_persisted_default_config_fields(self):
+        return ['project']
+
+    def create_default_repo_choice(self, default_repo):
+        try:
+            # default_repo should be the project_id
+            project = self.get_client().get_project(default_repo)
+        except (ApiError, ApiUnauthorized):
+            return ('', '')
+        return (project['id'], project['name'])
+
+    def get_project_choices(self, group, **kwargs):
```

Comment:
@markstory I debated whether to re-create this method or to adapt the existing `get_repository_choices`. Sadly I realized VSTS was more of an exception than I thought. Unlike GitLab it has both Repositories and Projects. Where a single project can have multiple repositories. But, workitems (issues in VSTS) are owned by the project and not individual repositories. x_x Meh.. 

What do you think of my decision here?

Reply:
I think you made the right choice here. Re-implementing the method is better than pushing vsts specific logic into the parent class.